admin/test/features/step_definitions/add_multiple_pupil_validation_steps.rb

```diff
@@ -0,0 +1,249 @@
+end
+
+Then(/^I can see the validation error for special charchter in first name and last name for multiple pupil upload$/) do
+  expect(@hashed_data[0][:errors].to_s.include?("First name can't contain special character")).to be_truthy, "Expected: 'First name can't contain special character'... But Got: #{@hashed_data[0][:errors].to_s}"
+  expect(@hashed_data[0][:errors].to_s.include?("Last name can't contain special characters")).to be_truthy, "Expected: 'Last name can't contain special characters'... But Got: #{@hashed_data[0][:errors].to_s}"
+  expect(@hashed_data[0][:errors].to_s.include?("Check the middle name does not contain special characters")).to be_truthy, "Expected: 'Check the middle name does not contain special characters'... But Got: #{@hashed_data[0][:errors].to_s}"
+
+end
+
+Then(/^I can see the validation error for wrong gender for multiple pupil upload$/) do
+  expect(@hashed_data[0][:errors].to_s.include?('Gender must be M or F')).to be_truthy, "Expected: 'Gender must be M or F'... But Got: #{@hashed_data[0][:errors].to_s}"
+end
+
+And(/^I can see the validation error for wrong month for multiple pupil upload$/) do
+  expect(@hashed_data[0][:errors].to_s.include?('Enter a valid month for date of birth')).to be_truthy, "Expected: 'Enter a valid month for date of birth'... But Got: #{@hashed_data[0][:errors].to_s}"
+end
+
+And(/^I can see the validation error for wrong day for multiple pupil upload$/) do
+  expect(@hashed_data[0][:errors].to_s.include?('Enter a valid day for date of birth')).to be_truthy, "Expected: 'Enter a valid day for date of birth'... But Got: #{@hashed_data[0][:errors].to_s}"
+end
+
+And(/^I can see the validation error for wrong year for multiple pupil upload$/) do
+  expect(@hashed_data[0][:errors].to_s.include?('Enter a valid year for date of birth')).to be_truthy, "Expected: 'Enter a valid year for date of birth'... But Got: #{@hashed_data[0][:errors].to_s}"
+end
+
+And(/^I can see the validation error for future date of birth for multiple pupil upload$/) do
+  expect(@hashed_data[0][:errors].to_s.include?("Date of birth can't be in the future")).to be_truthy, "Expected: 'Date of birth can't be in the future'... But Got: #{@hashed_data[0][:errors].to_s}"
+end
+
+And(/^I can see the validation error for duplicate UPN from attendance register for multiple pupil upload$/) do
+  expect(@hashed_data[0][:errors].to_s.include?('UPN is a duplicate of a pupil already in your register')).to be_truthy, "Expected: 'UPN is a duplicate of a pupil already in your register'... But Got: #{@hashed_data[0][:errors].to_s}"
+end
+
+# And(/^I can see the validation error for duplicate UPN from within the file for multiple pupil upload$/) do
+#   expect(@hashed_data[0][:errors].to_s.include?('More than 1 pupil record with same UPN')).to be_truthy, "Expected: 'More than 1 pupil record with same UPN'... But Got: #{@hashed_data[0][:errors].to_s}"
+# end
```

Comment:
@sinhaamit1980 Remove any commented code if its not needed

Reply:
Done